src/node_url.cc

```diff
@@ -75,6 +75,73 @@ void BindingData::Deserialize(v8::Local<v8::Context> context,
   CHECK_NOT_NULL(binding);
 }
 
+std::string EncodePathChars(const std::string& input_str, bool windows) {
+  std::ostringstream encoded;
```

Comment:
This is extremely slow. If you create a std::string view and call .reserve(origStrSize) and append to it while you iterate, it would be a lot faster.